buildSrc/src/main/kotlin/com/datadog/gradle/plugin/apisurface/CheckApiSurfaceChangesTask.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.gradle.plugin.apisurface
+
+import com.datadog.gradle.utils.execShell
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+open class CheckApiSurfaceTask : DefaultTask() {
+
+    lateinit var surfaceFile: File
+
+    init {
+        group = "datadog"
+        description = "Check the API surface of the library"
+    }
+
+    // region Task
+
+    @TaskAction
+    fun applyTask() {
+        val lines = project.execShell(
+            "git", "diff", "--color=never", "HEAD", "--", surfaceFile.absolutePath
+        )
+
+        val additions = lines.count { it.matches(Regex("^\\+[^+].*$")) }
+        val removals = lines.count { it.matches(Regex("^-[^-].*$")) }
+
+        if (additions > 0 || removals > 0) {
```

Comment:
This task already depends on the TASK_GEN_API_SURFACE meaning that if there's any change in the public api the _surfaceFile_ will change meaning that this condition:
`if(additions > 0 || removals > 0) ....` will be true and throw ISE. Is that what you want, for me it's a bit weird.

Reply:
No, the idea is to make sure that the apiSurface file was properly generated with the commit(s) in the branch.
So if the apiSurface is in sync with what's in the source code in the same commit, the newly generated file will be identical so there'll be no addition nor removals